### PR TITLE
Expose PVReclaimPolicy for TidbMonitor

### LINKED ISSUE #2329: monitor PVC will be deleted if TidbMonitor CR is deleted
## Bug Report

**What version of Kubernetes are you using?**

1.12.8
**What version of TiDB Operator are you using?**

v1.1.0-rc.2
**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

local-storage
**What's the status of the TiDB cluster pods?**

Running
**What did you do?**

Delete TidbMonitor CR
**What did you expect to see?**
monitor PVC remained
**What did you see instead?**
monitor PVC deleted

```
# k get pvc -n demo
NAME               STATUS   VOLUME              CAPACITY   ACCESS MODES   STORAGECLASS    AGE
demo-monitor       Bound    local-pv-45d26c43   440Gi      RWO            local-storage   37s
pd-demo-pd-0       Bound    local-pv-d097f77b   440Gi      RWO            local-storage   32s
tikv-demo-tikv-0   Bound    local-pv-3ac89acb   440Gi      RWO            local-storage   7s

# k delete tidbmonitor -n demo demo
tidbmonitor.pingcap.com "demo" deleted
# k get pvc -n demo           
NAME                   STATUS        VOLUME              CAPACITY   ACCESS MODES   STORAGECLASS    AGE
data0-demo-tiflash-0   Pending                                                     local-storage   1s
demo-monitor           Terminating   local-pv-45d26c43   440Gi      RWO            local-storage   56s
pd-demo-pd-0           Bound         local-pv-d097f77b   440Gi      RWO            local-storage   51s
tikv-demo-tikv-0       Bound         local-pv-3ac89acb   440Gi      RWO            local-storage   26s

# k get pvc -n demo 
NAME                   STATUS   VOLUME              CAPACITY   ACCESS MODES   STORAGECLASS    AGE
data0-demo-tiflash-0   Bound    local-pv-7855f795   440Gi      RWO            local-storage   78s
pd-demo-pd-0           Bound    local-pv-d097f77b   440Gi      RWO            local-storage   2m8s
tikv-demo-tikv-0       Bound    local-pv-3ac89acb   440Gi      RWO            local-storage   103s
```

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
close #2329 
Expose PVReclaimPolicy for TidbMonitor and e2e test for ensure it.

### Does this PR introduce a user-facing change?:

```release-note
Expose `PVReclaimPolicy` for `TidbMonitor` when storage is enabled.
```

### CODE REVIEW

File: pkg/monitor/monitor/monitor_manager.go [hunk 1]
```diff
@@ -95,6 +103,12 @@ func (mm *MonitorManager) Sync(monitor *v1alpha1.TidbMonitor) error {
 			return err
 		}
 		klog.V(4).Infof("tm[%s/%s]'s pvc synced", monitor.Namespace, monitor.Name)
+
+		// syncing all PVs managed by operator's reclaim policy to Retain
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/monitor_manager.go:107]: No need to be `Retain`.

Comment (Yisaer) [reply pkg/monitor/monitor/monitor_manager.go]: updated.

File: pkg/manager/meta/reclaim_policy_manager.go [hunk 2]
```diff
@@ -38,10 +41,26 @@ func NewReclaimPolicyManager(pvcLister corelisters.PersistentVolumeClaimLister,
...
+}
+
 func (rpm *reclaimPolicyManager) Sync(tc *v1alpha1.TidbCluster) error {
-	ns := tc.GetNamespace()
-	instanceName := tc.GetInstanceName()
+	return rpm.sync(tc.GetNamespace(), tc.GetInstanceName(), tc.IsPVReclaimEnabled(), tc.Spec.PVReclaimPolicy, tc)
+}
+
+func (rpm *reclaimPolicyManager) SyncMonitor(tm *v1alpha1.TidbMonitor) error {
+	return rpm.sync(tm.GetNamespace(), tm.GetName(), true, tm.Spec.PVReclaimPolicy, tm)
```
Comment (DanielZhangQD) [inline pkg/manager/meta/reclaim_policy_manager.go:60]:
```suggestion
	return rpm.sync(tm.GetNamespace(), tm.GetName(), false, tm.Spec.PVReclaimPolicy, tm)
```

Comment (DanielZhangQD) [reply pkg/manager/meta/reclaim_policy_manager.go:60]: If set to `true` and we set `label.AnnPVCDeferDeleting` to the monitor PVC later, the PVC will be removed after monitor scaling in.

**Pull request merged**